adventuregame.rb

```diff
@@ -1,73 +1,109 @@
+# frozen_string_literal: true
+
+# == Adventure \Game
+#
+# This is gonna be such a great game! Here's some details...
+# - TODO: add details
+#
+# == Running the game
+#
+# To run the game, ... (TODO: moar details needed; etc.)
+
 #GAME START
+
+# Player
+#
+# TODO: describe what this class is/does/is-responsible-for
+#
+# TODO: you might add an example of using this class here, too
 class Player
-  @hit_points
-  @attack_power
-  @chance_to_hit
-  
+  attr_accessor :hit_points
+  attr_accessor :attack_power
+  attr_accessor :chance_to_hit
+
   MAX_HIT_POINTS = 100
-  
+
+  # A predicate method to check whether a Player is alive or not
+  # - returns: Boolean (true/false)
   def alive?
     hit_points > 0
   end
-  
-  def attack_monster
-    #roll chance to hit
-    #compare chance to hit to monster stats
-    #deal damage / miss
+
+  # Performs an attack on the indicated target
+  #
+  # == Arguments
+  #
+  # +target+: +Monster+
+  # - an instance of a Monster that the player is attacking
+  def attack(target)
+    # 1. roll chance to hit
+    # 2. compare chance to hit to monster stats
+    # 3. deal damage / miss
   end
-  
-  def hurt_player(amount)
-    @hit_points = @hit_points - amount
+
+  # TODO: What's dis do?
+  def hurt_self(amount)
+    @hit_points -= amount
   end
-  
-  def heal_player(amount)
-    @hit_points = @hit_points + amount
+
+  # TODO: wats dat do?
+  def heal_self(amount)
+    @hit_points += amount
   end
-  
-    
-  
 end
 
+# Item
+# - describe what this class is/does/is-responsible-for
+# - docs/comments go here!
 class Item
 end
 
+
+# Monster
+# - describe what this class i/does/is-responsible-for
+# - docs/comments go here!
 class Monster
-  monsters = ["Rat", "Goblin", "Troll"]
-  @type
-  @chance_to_hit
-  @attack_power
-  @hit_points
-  
+  attr_accessor :hit_points
+  attr_accessor :attack_power
+  attr_accessor :chance_to_hit
+  attr_accessor :type
+
+  MONSTER_TYPES = %q[Rat Goblin Troll]
+
   def intitialize
-    monster = Monster.new
```

Comment:
I removed this line because its absolutely unnecessary; broken, even. The `#initialize` method is your "entry-point" into a class at run-time. You activate it (or "call" it) by calling the **Class'** `.new` method. Deep in the working's of Ruby's Object model, the `.new` method ends up calling the target Class' `#initialize` method. In other words, _SOMEWHERE ELSE_ in your code you will end up calling `monster = Monster.new`, and _THAT_ statement will end up calling this method to "initialize" the brand new object you `.new`ed.